models.go

```diff
@@ -1,21 +1,30 @@
 package rmb
 
-import "net/http"
+import (
+	"encoding/hex"
+	"fmt"
+	"net/http"
+
+	"github.com/pkg/errors"
+	"github.com/threefoldtech/substrate-client"
+)
 
 type Message struct {
-	Version    int    `json:"ver"`
-	ID         string `json:"uid"`
-	Command    string `json:"cmd"`
-	Expiration int64  `json:"exp"`
-	Retry      int    `json:"try"`
-	Data       string `json:"dat"`
-	TwinSrc    int    `json:"src"`
-	TwinDst    []int  `json:"dst"`
-	Retqueue   string `json:"ret"`
-	Schema     string `json:"shm"`
-	Epoch      int64  `json:"now"`
-	Proxy      bool   `json:"pxy"`
-	Err        string `json:"err"`
+	Version       int    `json:"ver"`
+	ID            string `json:"uid"`
+	Command       string `json:"cmd"`
+	Expiration    int64  `json:"exp"`
+	Retry         int    `json:"try"`
+	Data          string `json:"dat"`
+	TwinSrc       int    `json:"src"`
+	TwinDst       []int  `json:"dst"`
+	Retqueue      string `json:"ret"`
+	Schema        string `json:"shm"`
+	Epoch         int64  `json:"now"`
+	Proxy         bool   `json:"pxy"`
+	Err           string `json:"err"`
+	Signature     string `json:"sig"`
+	SignatureType string `json:"typ"`
```

Comment:
I know u wanna keep the 3 letters name for the attribute but if I look at the json alone `typ` is misleading because it does not tell type of what. Either we give it a better name, or I think it's better if the signature itself can have a type prefix. For example, assume signature is base64, a one letter prefix [0-9A-F] can be used to indicate the type of the signature.